components/admin_console/admin_sidebar_header/admin_sidebar_header.tsx

```diff
@@ -13,12 +12,16 @@ import Avatar from 'components/widgets/users/avatar';
 
 import AdminNavbarDropdown from 'components/admin_console/admin_navbar_dropdown';
 
-export default class SidebarHeader extends React.Component {
-    static propTypes = {
-        currentUser: PropTypes.object,
-    }
+type State = {
+    currentUser: {
```

Comment:
You don't need to define this type, we have a `UserProfile` type in `mattermost-redux` that covers this object.

Reply:
I had a crack at adding the UserProfile type from mattermost-redux but it doesn't have last_picture_update. Is there a way to change this without touching mattermost_redux?

Reply:
Also, figured out that the tests are failing because they don't take into account daylight savings time in New Zealand and are therefore showing the wrong date. I've change the two offending tests to NZ GMT and they pass without issue.

Reply:
> I had a crack at adding the UserProfile type from mattermost-redux but it doesn't have last_picture_update. Is there a way to change this without touching mattermost_redux?

Yes! You can define the type as such: `Partial<UserProfile>& {last_picture_update?: number}`

Reply:
Aha thank you for the, you taught me something new. I had to add id as well as it was throwing and error even though id is type string on the UserProfile object.

Reply:
Why `Partial`, i would expect to have the whole `UserProfile` + the `last_picture_update` field, but not a partial version of the `UserProfile`.

By the way, I think the correct approach is to add the `last_picture_update` as optional in mattermost-redux, and then use `UserProfile` directly here.

It is a longer path, but I think is the correct one.

Reply:
^ Agreed